### PR TITLE
Upgrade to Lucene 9.9.0

### PR DESCRIPTION
**Pull request by ChrisHegarty**
Upgrade to Lucene 9.9.0

TODO:
 - [x] Update TransportVersion constant ([75c5b87](https://github.com/elastic/elasticsearch/pull/102782/commits/75c5b870b4eb68fe0049a2e5f5395c938d031c11))
 - [x] Review and Update IndexVersion constant ([12906cd](https://github.com/elastic/elasticsearch/pull/102782/commits/12906cd8f33dae7c8d1169b3aa6d55f3f4bfe065))
 - [x] Remove lucene RC maven repo
 - [x] Verify Changelogs (there is currently just one!) ([36869d4](https://github.com/elastic/elasticsearch/pull/102782/commits/36869d417d54e8e19ccf5ec613468059e55cd958))

Note: This PR will merge it's commits into _main_ (rather than squashing), in order to preserve history and already-reviewed changes.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion NEW_SPARSE_VECTOR = def(8_500_001, Version.LUCENE_9_7_0);
     public static final IndexVersion SPARSE_VECTOR_IN_FIELD_NAMES_SUPPORT = def(8_500_002, Version.LUCENE_9_7_0);
     public static final IndexVersion UPGRADE_LUCENE_9_8 = def(8_500_003, Version.LUCENE_9_8_0);
-    public static final IndexVersion ES_VERSION_8_12 = def(8_500_004, Version.LUCENE_9_8_0);
+    public static final IndexVersion UPGRADE_LUCENE_9_9 = def(8_500_004, Version.LUCENE_9_9_0);
```
[the remark this answers is not in the document]
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/index/IndexVersions.java:92]: No objection from my side regarding this renaming

[the remark this answers is not in the document]
Comment (ChrisHegarty) [reply server/src/main/java/org/elasticsearch/index/IndexVersions.java:92]: Thanks @idegtiarenko. Rename and version bump :-)

**Comment by benwtrent** [comment]
@javanna you are correct, let me commit it really quick

**Comment by ChrisHegarty** [comment]
@elasticsearchmachine run elasticsearch-ci/part-1

**Review by benwtrent — APPROVED** [review]

**Pull request merged**